systemverilog-plugin/UhdmAst.cc

```diff
@@ -183,8 +183,75 @@ static std::string get_name(vpiHandle obj_h, bool prefer_full_name = false)
     } else if (auto s = vpi_get_str(last_check, obj_h)) {
         name = s;
     }
-    if (name.rfind('.') != std::string::npos) {
-        name = name.substr(name.rfind('.') + 1);
+    // We are looking for the ancestor name to use it as a delimeter
+    // when stripping the name of the current node.
+    // We used to strip the name by searching for "." in it, but this
+    // approach didn't work for the names whith "." as an escaped
+    // character.
+    vpiHandle parent_h = vpi_handle(vpiParent, obj_h);
+    if (parent_h) {
+        std::string parent_name;
+        if (auto p = vpi_get_str(vpiFullName, parent_h)) {
+            parent_name = p;
+        } else if (auto p = vpi_get_str(vpiName, parent_h)) {
+            parent_name = p;
+        } else if (auto p = vpi_get_str(vpiDefName, parent_h)) {
+            parent_name = p;
+        }
+        if (parent_name.empty()) {
+            // Nodes of certain types, like param_assign, don't have
+            // a name, so we need to look further for the ancestor.
+            parent_h = vpi_handle(vpiParent, parent_h);
+            if (parent_h) {
+                if (auto p = vpi_get_str(vpiFullName, parent_h)) {
+                    parent_name = p;
+                } else if (auto p = vpi_get_str(vpiName, parent_h)) {
+                    parent_name = p;
+                } else if (auto p = vpi_get_str(vpiDefName, parent_h)) {
+                    parent_name = p;
+                }
+                while (!parent_name.empty()) {
```

Comment:
I think this requires some refactoring as there is a lot of code repetition. Please extract recurrent patterns to separate functions.

Reply:
Done.